Новые цели для кражи - плутониевое ядро и срез СМ.

NightDawnFox opened the pull request:
<!-- Пишите **НИЖЕ** заголовков и **ВЫШЕ** комментариев, иначе ваш текст может не отобразиться. -->
<!-- В Contributing.MD вы можете найти некоторые рекомендации к оформлению пулл-реквеста. -->

## Описание
<!-- Опишите, что делает ваш ПР. Документировать каждую деталь не требуется, просто укажите основные изменения. -->
Добавляет две новые цели для кражи - плутониевое ядро и срез суперматерии.
В случае, если это ядро, то вору предоставляется пояснительная записка с инструкциями на русском языке и специализированная отвертка. Следуя инструкциям (лом, разварить, лом) он обнажает радиоактивное ядро боеголовки, которое будет излучать радиацию, пока не будет восстановлена герметичность боеголовки (вернуть на место вытащенную пласталь, заварить, восстановить). Далее, ядро боеголовки можно вытащить и поместить в контейнер, который уже и будет засчитываться как выполнение цели.
Важнейший момент. В случае кражи плутониевого ядра из боеголовки, ЕЕ НЕЛЬЗЯ БУДЕТ ВЗВЕСТИ. Если боеголовка взведена, вытащить ядро не получится. Заспавненные админами боеголовки будут с ядром.
В случае, если это срез суперматерии, то вору предоставляется записка с инструкциями, скальпель и щипцы. Следуя инструкциям, вор может срезать кусок суперматерии с кристалла, подобрать его щипцами и поместить в контейнер. Кусок суперматерии нельзя тащить, а попытка взятия его руками гибает вора. Само действие повреждает кристалл, что способно привести к его расслоению при плохой настройке СМ
Дополнительные правки с оффов:
-Контейнеры со срезом и ядром боеголовки можно открыть, если повредить их сильным взрывом
-Администрации отправляется оповещение о том, что боеголовку нельзя взвести (ее нет на уровне станции/ядро украдено)
-Если одновременно держать скальпель и щипцы, то срез суперматерии берется сразу
-Теперь при шифтклике на боеголовке пишется ее состояние и информация об этапах взлома боеголовки
-раундстартовые СЕ инженеры, атмостехи не смогут получить данную задачу ввиду легкости доступа к СМу, за исключением студентов.
## Ссылка на предложение/Причина создания ПР
<!-- Здесь оставьте ссылку на сообщение в #отчеты-по-предложениям, чтобы подтвердить, что ваше предложение одобрено. Либо укажите, почему этот ПР должен пройти без предложки. -->
<!-- Пример ссылки: https://discord.com/channels/617003227182792704/755125334097133628/ID-сообщения -->
https://discord.com/channels/617003227182792704/755125334097133628/1119081821204910140
## Демонстрация изменений
<!-- Здесь вы можете показать изменения внешне, к примеру новые спрайты, звуки или изменения карты. Или написать, что именно изменилось для игроков. Этот пункт полностью опционален и его можно удалить. -->
![image](https://github.com/ss220-space/Paradise/assets/116907157/4789b700-ca7d-4611-80b1-c5086f60ddd7)
https://www.youtube.com/watch?v=u0nOM_7aau8

NightDawnFox committed:
test 1

NightDawnFox committed:
Прокреф

Однажды зюзя не проебался с проками в своем ПРе и на планете вылечили все болезни

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into porting-port-from-TG

NightDawnFox committed:
Патч в связи с рефактором

NightDawnFox committed:
huge update from offs

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into porting-port-from-TG

NightDawnFox committed:
fixing bugs

NightDawnFox committed:
Больше радейки, меньше багов

NightDawnFox committed:
Update supermatter.dm

NightDawnFox committed:
Merge branch 'master220' into porting-port-from-TG

NightDawnFox committed:
major SM update

NightDawnFox committed:
Подгонка цифр СМ под реалии

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into supermatter-sliver-and-nuke-core

NightDawnFox committed:
update

Все распыления от осколка заменены на гиб, обновлен код плутониевого ядра после рефактора от Звея, дополнены инструкции

NightDawnFox committed:
Update theft_items.dm

этого тут быть не должно...

Rerik007 commented on code/game/objects/items/theft_items.dm:
```diff
@@ -0,0 +1,380 @@
+	core = new_core
+	icon_state = "core_container_loaded"
+	to_chat(user, "<span class='warning'>Container is sealing...</span>")
+	addtimer(CALLBACK(src, PROC_REF(seal)), 100)
```
```suggestion
	addtimer(CALLBACK(src, PROC_REF(seal)), 10 SECONDS)
```
Раз уж внизу также

Rerik007 commented on code/game/objects/items/theft_items.dm:
```diff
@@ -0,0 +1,380 @@
+/obj/item/nuke_core_container/proc/seal()
+	if(!QDELETED(core))
+		STOP_PROCESSING(SSobj, core)
+		ADD_TRAIT(core, TRAIT_BLOCK_RADIATION, src)
```
Просто интересно, только у этого контейнера добавляется трейт защиты оттого что у суперматерии радиация выше и её там не остановить?

NightDawnFox replied:
Нет, это мой косяк. Не усмотрел. Если оно будет постоянно фонить на 40 рада, то цель немного невыполнима. Сейчас поправлю

NightDawnFox replied:
Добавил нужные трейты, проверил на локалке, больше не фонит

Rerik007 commented on code/modules/power/supermatter/supermatter.dm:
```diff
@@ -372,6 +373,39 @@
+						to_chat(user, "<span class='boldwarning'>A tiny piece of [W] falls off, rendering it useless!</span>")
+					var/obj/item/nuke_core/supermatter_sliver/S = new /obj/item/nuke_core/supermatter_sliver(drop_location())
+
+					var/obj/item/retractor/supermatter/tongs = M.get_inactive_hand(/obj/item/retractor/supermatter)
```
Нет, get_inactive_hand не ищет по типу, поэтому нужна проверка что это именно ретрактор. А то будет рантаймить, пусть и без последствий.

В данном случае что-то наподобие такого можно попробовать: 
```suggestion
					var/obj/item/retractor/supermatter/tongs = M.get_inactive_hand()
					if(!istype(tongs)
						return
```
кхм, скобку забыл здесь я, извиняюсь///

Rerik007 commented on code/game/objects/items/theft_items.dm:
```diff
@@ -0,0 +1,380 @@
+	for(var/mob/living/L in view(5, src))
+		L.apply_effect(60, IRRADIATE)
+	playsound(src, 'sound/effects/supermatter.ogg', 50, TRUE)
+	QDEL_NULL(sliver)
```
В данном случае ретрактор не уничтожается, поэтому бы стоит обновлять его иконку на пустую.

NightDawnFox replied:
Так лучше?

Rerik007 replied:
Наверное да... Возможно лучше бы было сделать эти обновления иконок отдельным update_icon проком и вызывать его тут. Впрочем думаю это необязательно. 👍

Rerik007 reviewed (commented):
Наверное будет забавно смотреть, как цель на см делают на какой-нибудь коробке или в раунде с ранним подрывом.

Rerik007 reviewed (commented):
Хм. И зачем ещё в спрайтах:
```
Nuke Container + Core
Nuke Core Container Overlay 
Nuke Core Container Pulsing Overlay
```
Они для того чтобы их потом было легче редактировать и доставать что-то или просто?

NightDawnFox committed:
Apply suggestions from code review

Co-authored-by: Rerik007 <120549107+Rerik007@users.noreply.github.com>

NightDawnFox commented:
@Rerik007, это файлы с ТГ. В принципе их можно удалить, но если кто-то что-то будет делать на основе этих коробок и оверлеев (как например на йогах сделали осколок антиноблиума, что СМ уничтожает), то спрайтерам будет удобнее, если оверлеи останутся

NightDawnFox committed:
Обновление кода в связи с ревью

Исправление небольшого недочета, добавление трейтов защиты от радиации для контейнера осколка суперматерии, обновление иконок щипцов

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into supermatter-sliver-and-nuke-core

NightDawnFox committed:
Merge remote-tracking branch 'upstream/master220' into supermatter-sliver-and-nuke-core
